docs/source/configuration.rst

```diff
@@ -295,6 +295,26 @@ For the compiled contracts you will now see the following in the output:
   }
 
 
+typechain
+^^^^^^^^^
+
+Waffle supports `typechain <https://github.com/ethereum-ts/TypeChain>`__ artifacts generation.
+To enable typed artifacts generation you should set :code:`enable` property to :code:`true`.
+You are also able to define target folder for your artifacts by defining :code:`outputDir` property, which is set to :code:`./types` by default.
+Property :code:`outputDir` is a path relative to :ref:`outputDirectory`.
+
+Example:
+
+.. code-block:: json
+
+  {
+    "typechain": [
+      "enable": true,
```

Comment:
Why not support by default?

Reply:
Could be.

Note: just noticed I need to update docs, because we flattened the config options for typechain.

Reply:
Apparently TypeChain is not compatible with KLAV
```
 |   1) E2E: Solidity 5 - KLAB
 |        compiles without errors:
 |      Error: Not a valid ABI
 |       at Object.extractAbi (/Users/krzysztof/Workspace/Waffle/node_modules/typechain/src/parser/abiParser.ts:308:9)
 |       at Ethers.transformAbiOrFullJsonFile (/Users/krzysztof/Workspace/Waffle/node_modules/@typechain/ethers-v5/src/index.ts:78:17)
 |       at Ethers.transformFile (/Users/krzysztof/Workspace/Waffle/node_modules/@typechain/ethers-v5/src/index.ts:56:17)
 |       at TypeChain.transformFile (/Users/krzysztof/Workspace/Waffle/node_modules/typechain/src/TypeChain.ts:58:26)
 |       at Object.tsGenerator (/Users/krzysztof/Workspace/Waffle/node_modules/ts-generator/dist/tsGenerator.js:29:51)
 |       at Object.generateTypes (src/generateTypes.ts:7:3)
 |       at Object.compileProject (src/compiler.ts:15:5)
 |       at Context.<anonymous> (test/compiler/e2e.ts:71:9)
```